rusoto/credential/src/instance_metadata.rs

```diff
@@ -40,6 +40,7 @@ const AWS_CREDENTIALS_PROVIDER_PATH: &str = "latest/meta-data/iam/security-crede
 pub struct InstanceMetadataProvider {
     client: HttpClient,
     timeout: Duration,
+    port_for_testing: Option<String>,
```

Comment:
This should be done better. My goal is to prevent complete overriding of the instance profile source to defend against potentially dangerous or unwanted sources. Is this actually a problem though? The [container provider](https://github.com/rusoto/rusoto/blob/master/rusoto/credential/src/container.rs#L22) lets you do whatever, following other SDKs.